rep-0145.rst

```diff
@@ -0,0 +1,203 @@
+REP: 145
+Title: Conventions for IMU Sensor Drivers
+Author: Paul Bovbel <pbovbel@clearpathrobotics.com>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 02-Feb-2015
+Post-History: 02-Feb-2015
+
+
+Abstract
+========
+
+This REP defines common parameters, topics, namespaces, and data processing conventions for drivers of Inertial Measurement Unit (IMU) sensors. This includes accelerometers, gyroscopes, magnetomers, and any combination thereof.
+
+Motivation
+==========
+
+This REP seeks to standardize an interface for IMU drivers in the ROS ecosystem, providing for IMU-type sensors with a variety of capabilities and conventions. By formalizing a common approach based on existing driver implementations, this REP serves as a guideline for new driver development, while minimizing the changes required to non-conforming implementations. This REP discourages the in-driver manipulation of sensor data, by specifying the conditions where sensor data will be properly handled by ROS transform and coordinate specifications. The REP also specifies fallback approaches to correcting sensor data that does not conform to ROS conventions.
+
+Specification
+=============
+
+Common Parameters
+-----------------
+
+IMU driver implementations should read as many of these parameters as are relevant.
+
+* `~port` (`string`)
+
+  - Represents the serial port used to connect to the device.
+
+* `~baud` (`int`)
+
+  - Represents the baud rate of the serial link.
+
```

Comment:
I think the parameters `port` and `baud` should be removed. It's specific to serial port communication (as opposed to USB, for example). Same for `autocalibrate`.

Also, I would move the whole "common parameters" paragraph way down, since it's the least important bit of this REP. What's important for interoperability is the semantics of what's published on the topics, not driver parameters. These are implementation specific, and if one switches the IMU, one has to switch the driver as well, so having to use a different set of parameters for the driver is okay.